### PR TITLE
Allow Float and BaseFloat trait types to accept longs.

### LINKED ISSUE #271: Float trait type should accept longs.
On Python 2, the `Float` trait type should accept objects of Python type `long`.

```
>>> from traits.api import HasTraits, Float
>>> class A(HasTraits):
...     foo = Float
... 
>>> a = A()
>>> a.foo = 2
>>> a.foo = 2L
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mdickinson/Enthought/ETS/traits/traits/trait_handlers.py", line 172, in error
    value )
traits.trait_errors.TraitError: The 'foo' trait of an A instance must be a float, but a value of 2L <type 'long'> was specified.
>>> a.foo = 2**64
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mdickinson/Enthought/ETS/traits/traits/trait_handlers.py", line 172, in error
    value )
traits.trait_errors.TraitError: The 'foo' trait of an A instance must be a float, but a value of 18446744073709551616L <type 'long'> was specified.
```

### PR DESCRIPTION
**Pull request by mdickinson**
Fixes #271.

### CODE REVIEW

File: traits/tests/test_float.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# -----------------------------------------------------------------------------
+#
+#  Copyright (c) 2013, Enthought, Inc.
```
Comment (rkern) [inline traits/tests/test_float.py]: Lies!

File: traits/tests/test_float.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        a = self.test_class()
+        a.value = 2
+        self.assertIs(type(a.value), float)
+        self.assertEqual(a.value, 2.0)
+
+    @unittest.skipUnless(sys.version_info < (3,), "Not applicable to Python 3")
+    def test_accepts_small_long(self):
+        a = self.test_class()
+        # Value large enough to be a long on Python 2.
+        a.value = long(2)
```
Comment (rkern) [inline traits/tests/test_float.py]: I think that comment was intended to go on the next test.

**Comment by rkern** [comment]
Comment nits. Otherwise, LGTM.

File: traits/tests/test_float.py (hunk 1 above)
Comment (mdickinson) [inline traits/tests/test_float.py]: :rat:s.  Thanks.

**Pull request merged**